docs/hosting-a-repository.rst

```diff
@@ -143,34 +142,38 @@ The instructions will use Gitlab.com.
     variables:
       # Stable Flathub repo
       RUNTIME_REPO: "https://flathub.org/repo/flathub.flatpakrepo"
-    before_script:
+    script:
+      - |
+        cat <<EOF > /etc/passwd
+        root:x:0:0:root:/root:/bin/bash
+        EOF
+
+        cat <<EOF > /etc/group
+        root:x:0:
+        EOF
+
```

Comment:
Do you know why this is necessary?

Reply:
fdsdk's images on docker.io, don't have a root user configured in it. The flathub actions images https://github.com/flathub-infra/flatpak-github-actions/pkgs/container/flatpak-github-actions are derived from the linter image which uses fdsdk as base

It silences a bunch of warnings about root user.

Reply:
Is that a purposeful choice by fdsdk? It just seems odd to me.

Reply:
Only the ones on docker.io for some reason, the ones on gitlab registry have the root user in them.

I was talking to @nanonyme about this a few months back. Can't remember the exact reason.

Reply:
@TingPing So the docker.io images are being pushed from https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/.gitlab-ci.yml?ref_type=heads#L914-931 which does not include any root configuration

while the gitlab registry image is from https://gitlab.com/freedesktop-sdk/infrastructure/freedesktop-sdk-docker-images/

which has this element to add the users https://gitlab.com/freedesktop-sdk/infrastructure/freedesktop-sdk-docker-images/-/blob/61d17ed68a166010297f135293439a81bc3ee8ca/elements/extra-fs.bst

The main repo does not have a separate element like that, it uses a script for setting up root user in the VM images, but not in OCI.

Reply:
We would probably add the extra-fs.bst element in some form to the main repo.